src/inference/smc.js

```diff
@@ -71,6 +79,20 @@ module.exports = function(env) {
     return this.runCurrentParticle();
   };
 
+  // Error function for error handling
+  // this.throwOnError is true: directly throw error
+  // this.throwOnError is false: return error (string) as infer result
+  SMC.prototype.error = function(errType) {
+    if (this.throwOnError) {
+      throw new Error(errType);
+    } else {
+      this.err = errType;
+      // the way to return differs from enumerate and rejection because of CPS
+      // directly calling this.k doesn't work
+      return this.finish();
```

Comment:
I'd like to recreate the problem, but when I tried changing `this.finish()` to `this.k(this.s, errType)` then all the tests passed, and writing test programs to generate errors behaved as I expected. Any idea what I'm missing?

(I think all the places we call `this.error` in SMC are in CPS already, so I don't think it's the same problem as `getSupport`, but I could be mistaken.)